iceoryx_hoofs/include/iceoryx_hoofs/internal/graphs/directed_graph.hpp

```diff
@@ -28,10 +28,10 @@ template <typename VertexType, int32_t VERTEX_LIMIT, int32_t DEGREE_LIMIT>
 class DirectedGraph
 {
   public:
-    using Index_t = int32_t;
+    using Index_t = uint64_t;
     using AdjacencyList = iox::cxx::vector<VertexType*, DEGREE_LIMIT>;
 
-    static constexpr Index_t INVALID_INDEX = -1;
+    static constexpr Index_t INVALID_INDEX = std::numeric_limits<uint64_t>::max();
```

Comment:
I will reduce the size to 32 but would stick with unsigned since this type is used in array operator[] and this requires an unsigned integer. So I have to write a little less static_cast